server_addon/houdini/client/ayon_houdini/api/hda_utils.py

```diff
@@ -0,0 +1,498 @@
+    # bypass state of the node.
+    parent = node.parent()
+    images = lib.get_background_images(parent)
+    if not images:
+        return
+
+    brightness = 0.3 if node.isBypassed() else 1.0
+    has_changes = False
+    node_path = node.path()
+    for image in images:
+        if image.relativeToPath() == node_path:
+            image.setBrightness(brightness)
+            has_changes = True
+
+    if has_changes:
+        lib.set_background_images(parent, images)
+
+
+def keep_background_images_linked(node, old_name):
+    """Reconnect background images to node from old name.
+
+     Used as callback on node name changes to keep thumbnails linked."""
+    from ayon_houdini.api.lib import (
+        get_background_images,
+        set_background_images
+    )
+
+    parent = node.parent()
+    images = get_background_images(parent)
+    if not images:
+        return
+
+    changes = False
+    old_path = f"{node.parent().path()}/{old_name}"
+    for image in images:
+        if image.relativeToPath() == old_path:
+            image.setRelativeToPath(node.path())
+            changes = True
+
+    if changes:
+        set_background_images(parent, images)
+
+
+def select_folder_path(node):
+    """Show dialog to select folder path.
+
+    When triggered it opens a dialog that shows the available
+    folder paths within a given project.
+
+    Note:
+        This function should be refactored.
+        It currently shows the available
+          folder paths within the current project only.
+
+    Args:
+        node (hou.OpNode): The HDA node.
+    """
+    main_window = lib.get_main_window()
+
+    project_name = node.evalParm("project_name")
+    folder_path = node.evalParm("folder_path")
+
+    class PickDialog(QtWidgets.QDialog):
+        def __init__(self, parent=None):
+            super(PickDialog, self).__init__(parent)
+            # TODO: Add pick project field
+            self.setWindowTitle("Select folder..")
+
+            filter_widget = QtWidgets.QLineEdit()
+            filter_widget.setPlaceholderText("Filter folders..")
+            folder_widget = SimpleFoldersWidget(parent=self)
+            accept_button = QtWidgets.QPushButton("Accept")
+
+            layout = QtWidgets.QVBoxLayout(self)
+            layout.addWidget(filter_widget)
+            layout.addWidget(folder_widget)
+            layout.addWidget(accept_button)
+
+            filter_widget.textChanged.connect(folder_widget.set_name_filter)
+
+            folder_widget.double_clicked.connect(self.on_confirm)
+            accept_button.clicked.connect(self.on_confirm)
+
+            self.folder_widget = folder_widget
+
+        def get_selected_folder_path(self):
+            return self.folder_widget.get_selected_folder_path()
+
+        def on_confirm(self):
+            self.close()
+
+    # Note: The following dialog doesn't support changing `the project_name`
+    #         But, having a semi-functional dialog is better than nothing.
```

Comment:
Should this TODO removed ?  I can change the project.
```suggestion
```

![image](https://github.com/ynput/ayon-core/assets/20871534/96c796b8-34fd-420a-b6da-9240b9b470d8)

Reply:
Yes.